firebase-firestore/src/test/java/com/google/firebase/firestore/remote/BloomFilterTest.java

```diff
@@ -0,0 +1,252 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.remote;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.stream.Stream;
+import org.json.JSONObject;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(manifest = Config.NONE)
+public class BloomFilterTest {
+  private static final String GOLDEN_DOCUMENT_PREFIX =
+      "projects/project-1/databases/database-1/documents/coll/doc";
+  private static final String GOLDEN_TEST_LOCATION =
+      "src/test/resources/bloom_filter_golden_test_data/";
+
+  @Test
+  public void instantiateEmptyBloomFilter() {
+    BloomFilter bloomFilter = new BloomFilter(new byte[0], 0, 0);
+    assertEquals(bloomFilter.getBitCount(), 0);
+  }
+
+  @Test
+  public void instantiateNonEmptyBloomFilter() {
+    BloomFilter bloomFilter1 = new BloomFilter(new byte[] {1}, 0, 1);
```

Comment:
For tests like `instantiateNonEmptyBloomFilter()`, which effectively test more than one thing, separate each test into its own logic scope using `{` and `}`. This provides several benefits, including

1. It logically groups the two tests together, improving readability.
2. It prevents accidental usage of variables from one test in another, improving robustness.
3. Variable names can be re-used, increasing the symmetry between the tests, making it easier for a human to see what's the same and what's different between the cases.

For example, the body of `instantiateNonEmptyBloomFilter()` would become this:

```java
{
  BloomFilter bloomFilter = new BloomFilter(new byte[] {1}, 0, 1);
  assertEquals(bloomFilter.getBitCount(), 8);
}
{
  BloomFilter bloomFilter = new BloomFilter(new byte[] {1}, 7, 1);
  assertEquals(bloomFilter.getBitCount(), 1);
}
```

Reply:
Please make this change in `constructorShouldThrowNPEOnNullBitmap()`, `constructorShouldThrowIAEOnNegativePadding()`, `constructorShouldThrowIAEOnNegativeHashCount()`, `mightContainWithEmptyStringMightReturnFalsePositiveResult()`, and `bloomFilterToString()` too.